client/ayon_core/hosts/houdini/plugins/publish/validate_review_colorspace.py

```diff
@@ -33,6 +33,13 @@ class ValidateReviewColorspace(pyblish.api.InstancePlugin,
 
     def process(self, instance):
 
+        rop_node = hou.node(instance.data["instance_node"])
+
+        if rop_node.type().name() != "opengl":
+            self.log.debug("Skipping Validation. Rop node {} "
+                           "is not an OpenGl node.".format(rop_node.path()))
+            return
```

Comment:
Your comment is so true. 
The thing I'm confused about how to add mark an instance as reviewable. 
as far as I know, I need to 
1. append `"review"` to `instance.data["families"]`
2. append `"review"` to `tags` in the `representation`

And, this validator is triggered whenever appending `"review"` to families. which is as you said: *this validator is now just running over the wrong instances*

> Also, when searching the code base, I found that sometimes we `instance.data.update({"review" : True})` and I'm not sure how this works.

Reply:
Note that:

```python
instance.data.update({"review" : True})
```

Is the same as:
```
instance.data["review"] = True
```

Because `instance.data` is just a `dict` and thus:
```python
new = {"review": True}
instance.data.update(new)
```
Is basically the same.

---



> as far as I know, I need to
> 
>     1. append `"review"` to `instance.data["families"]`
> 
>     2. append `"review"` to `tags` in the `representation`

How "marking something for review" is confusing has been mentioned quite a few times before. I personally don't know all too much about it and potentially @jakubjezek001 or @antirotor are more knowledgeable about the reasoning or exact steps needed to "mark for review" which I think for you just means to also get burnins and previews generated, yes? I know @iLLiCiTiT did some work on the burning/review logic so might be able to chime in as well.

Reply:
`instance.data["review"] = True` is saying: when this instance, or derived instance is processed on farm, it will add "review" **family** so it will be processed by *Extract Review*. Mostly Deadline based logic that we'll be happy to refactor.

Reply:
Should I then add the render product types by default to this plugin + add something like ?
```python
if not instance.data["review"]:
    # skipping
    return
```
https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/hosts/houdini/plugins/publish/collect_review_data.py

Reply:
to make it more complicated, you need to take care wheter instance is actually reviewable -> not all renders are - for example if you have instance, that is producing normal, point position and depth renders, there is nothing you can use for reviews.

Reply:
> to make it more complicated, you need to take care wheter instance is actually reviewable -> not all renders are 

Revisiting the AOVs filters.
settings location: 
- `ayon+settings://deadline/publish/ProcessSubmittedJobOnFarm/aov_filter` 

In the current state of this PR, I found out that these filters work with Houdini farm publishing. 
![image](https://github.com/ynput/ayon-core/assets/20871534/cd9a3865-ffcc-4c40-a183-967980216a62)

But, I faced some problems because of the regex filter doesn't match the default Houdini behavior. 
Because , by default Not all ROPs support `$AOV` in render paths.
So, I actually had to add AOV names manually in the mantra rop
![image](https://github.com/ynput/ayon-core/assets/20871534/83402a38-2a32-4374-b073-063cb81b2c2c)

Every product type can have its special cases. 
For example: Redshift 
It has `$AOV` which is cool but once I enable `RS_aovMultipart` and add cryptomatte AOV
in that case: redshift will render cryptomatte separately and ignore `beauty` AOV name suffix thus no review will be created as `beauty` will no longer exist in the file name.

Reply:
>For example: Redshift
It has $AOV which is cool but once I enable RS_aovMultipart and add cryptomatte AOV
in that case: redshift will render cryptomatte separately and ignore beauty AOV name suffix thus no review will be created as beauty will no longer exist in the file name.

I've just discovered that I should add `renderer` key in instance data 
which solves redshift issue mentioned above. 

https://github.com/ynput/ayon-core/blob/27dea6c97a5bac4fc5ba7ca4a0932f4d6a993109/client/ayon_core/pipeline/farm/pyblish_functions.py#L633-L637

Reply:
> Hmm - yeah. This doesn't feel entirely right. I feel like this validator is now just running over the wrong instances - which should be solved in a way that it just does not run to begin with when it shouldn't. I understand why it's done now, since you're likely adding "review" family to something that isn't actually an OpenGL review instance - but maybe the regular review instances should instead have `opengl_rop` family or whatever and have these validators target that?

I added a TODO about it. I think it should be fixed in a separate PR.
Therefore, I'm marking this as resolved.